### DISCUSSION TITLE
npm install errors

### DISCUSSION THREAD

**Discussion by zunder1990** [pretalx/pretalx#1597] [General]
I am trying to get pretalx up and going for my linux event. Fresh install of Debian 11. 

 I am at step "python3 -m pretalx rebuild"

`
> @pretalx/schedule-editor@0.0.0 build
> vite build
file:///var/pretalx/.local/lib/python3.9/site-packages/pretalx/frontend/schedule-editor/node_modules/vite/bin/vite.js:7
    await import('source-map-support').then((r) => r.default.install())
    ^^^^^
SyntaxError: Unexpected reserved word
    at Loader.moduleStrategy (internal/modules/esm/translators.js:133:18)
    at async link (internal/modules/esm/module_job.js:42:21)
npm ERR! code 1
npm ERR! path /var/pretalx/.local/lib/python3.9/site-packages/pretalx/frontend/schedule-editor
npm ERR! command failed
npm ERR! command sh -c vite build
npm ERR! A complete log of this run can be found in:
npm ERR!     /var/pretalx/.npm/_logs/2023-10-04T10_42_00_357Z-debug.log
Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/var/pretalx/.local/lib/python3.9/site-packages/pretalx/__main__.py", line 9, in <module>
    execute_from_command_line(sys.argv)
  File "/var/pretalx/.local/lib/python3.9/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/var/pretalx/.local/lib/python3.9/site-packages/django/core/management/__init__.py", line 436, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/var/pretalx/.local/lib/python3.9/site-packages/django/core/management/base.py", line 412, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/var/pretalx/.local/lib/python3.9/site-packages/django/core/management/base.py", line 458, in execute
    output = self.handle(*args, **options)
  File "/var/pretalx/.local/lib/python3.9/site-packages/pretalx/common/management/commands/rebuild.py", line 54, in handle
    subprocess.check_call(["npm", "run", "build"], cwd=frontend_dir, env=env)
  File "/usr/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['npm', 'run', 'build']' returned non-zero exit status 1.
pretalx@xxxxxx:~$ cat /var/pretalx/.npm/_logs/2023-10-04T10_42_00_357Z-debug.log
0 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'run', 'build' ]
1 info using npm@7.5.2
2 info using node@v12.22.12
3 timing config:load:defaults Completed in 2ms
4 timing config:load:file:/usr/share/nodejs/npm/npmrc Completed in 2ms
5 timing config:load:builtin Completed in 2ms
6 timing config:load:cli Completed in 1ms
7 timing config:load:env Completed in 1ms
8 timing config:load:file:/var/pretalx/.local/lib/python3.9/site-packages/pretalx/frontend/schedule-editor/.npmrc Completed in 1ms
9 timing config:load:project Completed in 1ms
10 timing config:load:file:/var/pretalx/.npmrc Completed in 0ms
11 timing config:load:user Completed in 0ms
12 timing config:load:file:/etc/npmrc Completed in 0ms
13 timing config:load:global Completed in 0ms
14 timing config:load:cafile Completed in 0ms
15 timing config:load:validate Completed in 0ms
16 timing config:load:setUserAgent Completed in 1ms
17 timing config:load:setEnvs Completed in 1ms
18 timing config:load Completed in 9ms
19 verbose npm-session c5ee7d3d86f9dfb5
20 timing npm:load Completed in 16ms
21 timing command:run-script Completed in 98ms
22 verbose stack Error: command failed
22 verbose stack     at ChildProcess.<anonymous> (/usr/share/nodejs/@npmcli/promise-spawn/index.js:64:27)
22 verbose stack     at ChildProcess.emit (events.js:314:20)
22 verbose stack     at maybeClose (internal/child_process.js:1022:16)
22 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)
23 verbose pkgid @pretalx/schedule-editor@0.0.0
24 verbose cwd /var/pretalx/.local/lib/python3.9/site-packages/pretalx/frontend/schedule-editor
25 verbose Linux 5.10.0-25-amd64
26 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "build"
27 verbose node v12.22.12
28 verbose npm  v7.5.2
29 error code 1
30 error path /var/pretalx/.local/lib/python3.9/site-packages/pretalx/frontend/schedule-editor
31 error command failed
32 error command sh -c vite build
33 verbose exit 1 
`

**Comment by rixx** [comment]
`SyntaxError: Unexpected reserved word` when encountering `await` sounds like a very old node version that pretalx is not compatible with. Off the top of my head I'd say something like 16+ should be good.

**Comment by zunder1990** [comment]
Thanks that was it.
Here is what I did to fix it
```
npm cache clean -f
npm install -g n
sudo n stable
```

versions is now v18.18.0
